### PR TITLE
Add install generator

### PR DESCRIPTION
**Pull request by iseth**
# 🚀 Add RBUI Install Generator

## 🌟 Features

This PR introduces a new Rails generator for seamless installation of RBUI components and configuration. With this update, users can quickly set up a new Rails application with RBUI integration.

### 🛠️ New Generator: `rbui:install`

The new generator performs the following tasks:

- 📦 Adds RBUI gem to the project
- 🏗️ Sets up necessary configurations
- 🎨 Installs and configures Tailwind CSS
- 🔗 Configures JavaScript dependencies

## 🚀 Usage

To use this new feature:

1. Set up a new Rails application
2. Add the following to your Gemfile:
```ruby
gem "rbui", path: "../rbui"
```
3. Run the generator:
```bash
rails g rbui:install
```

## 📝 Notes
This generator aims to provide a smooth onboarding experience for new RBUI users. Feedback and suggestions for improvement are welcom

### CODE REVIEW

**Comment by iseth** [comment]
@cirdes don't squash these changes and also don't merge #97. or just review the pr's and i can merge them.

File: lib/generators/rbui/install/templates/application.tailwind.css.tt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+@tailwind base;
```
Comment (cirdes) [inline lib/generators/rbui/install/templates/application.tailwind.css.tt:1]: you should use the variables on this file: https://github.com/PhlexUI/PhlexUI-web/blob/v1/app/assets/stylesheets/application.tailwind.css

File: lib/generators/rbui/install/templates/tailwind.config.js.tt [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// For importing tailwind styles from phlex_ui/phlex_ui_pro gem
+const execSync = require('child_process').execSync;
```
Comment (cirdes) [inline lib/generators/rbui/install/templates/tailwind.config.js.tt:2]: same here: https://github.com/PhlexUI/PhlexUI-web/blob/v1/tailwind.config.js

File: lib/rbui/railtie.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        initializer "rbui.set_generator_namespace" do
+          Rails::Generators.namespace(RBUI::Generators, as: "rbui")
+        end
+      end
+
+      # Add Zeitwerk configuration
+      initializer "rbui.configure_zeitwerk" do
+        Rails.autoloaders.main.inflector.inflect(
+          "phlex_ui" => "PhlexUI",
```
Comment (cirdes) [inline lib/rbui/railtie.rb:19]: Almost 50% of the project are using rbui module and I want to have all of them in a couple of weeks. Maybe you should remove all "phlex_ui" references.

**Review by cirdes — COMMENTED** [review]
Very excited with this PR!

**Comment by iseth** [comment]
wait for rest of components to be migrated.

**Comment by iseth** [comment]
this is part of #56

**Comment by iseth** [comment]
Ok this is completely ready for final code review. @cirdes

**Review by iseth — COMMENTED** [review]
looks good to me

**Pull request merged**